usersession/agent/session_agent.go

```diff
@@ -165,6 +187,23 @@ func (l *closeOnceListener) Close() error {
 }
 
 func (s *SessionAgent) Init() error {
+	// Set up D-Bus connection
+	var err error
+	s.bus, err = dbusSessionBus()
```

Comment:
Nitpick, it might be prudent to split Init() to call into smaller helpers that initialize the specific connection type, this would also simplify things like error handling as you could then do:
```
bus, err := dbusSessionBus()
if err != nil {
   logger.Noticef("...")
   return nil
}
// set up the rest
s.bus = bus
return nil
```

Right now that notice is at the bottom of the logic